src/main/scala/firrtl/backends/verilog/VerilogEmitter.scala

```diff
@@ -482,6 +489,15 @@ class VerilogEmitter extends SeqTransform with Emitter {
     def getConnectEmissionOption(target: ReferenceTarget): ConnectEmissionOption =
       connectEmissionOption(target)
 
+    // Defines the memory initialization based on the annotation
+    // Defaults to having the memories inside the `ifndef SYNTHESIS` block
+    def emitMemoryInitAsNoSynth: Option[Boolean] = {
+      annotations.collectFirst {
+        case MemoryNoSynthInit => Option(true)
+        case MemorySynthInit   => Option(false)
+      }.getOrElse(Option(true))
+    }
```

Comment:
You're always making sure it's `Some` anyway, might as well just return the `Boolean`.
```suggestion
    def emitMemoryInitAsNoSynth: Boolean = {
      annotations.collectFirst {
        case MemoryNoSynthInit => true
        case MemorySynthInit   => false
      }.getOrElse(true)
    }
```

Reply:
Also, I'm ambivalent if we provide just 1 annotation or both, but if we provide both, we need to check somewhere that the user doesn't provide both. It's not great waiting till the VerilogEmitter to check, but it's also possible that a custom transform could erroneously inject one that contradicts the one provided by the user so this is probably the right place to check.

Reply:
Don't know why it was complaining and wanted an Option[Bool] before.
Fixed.